patches/minecraft/net/minecraft/client/Minecraft.java.patch

```diff
@@ -113,7 +113,36 @@
      }
  
      public boolean func_152349_b()
-@@ -955,11 +967,6 @@
+@@ -770,8 +782,20 @@
+         return this.field_135017_as.func_135042_a() || this.field_71474_y.field_151455_aw;
+     }
+ 
++    @Deprecated // Forge: Use selective scheduleResourceRefresh method
+     public void func_110436_a()
+     {
++        refreshResources(net.minecraftforge.client.resource.ReloadRequirements.all()); // Forge: Delegate to selective loader and load everything.
```

Comment:
Looking at this deeper, it may be cleaner to move these new `refreshResources` methods into a Forge class like `FMLClientHandler`.
Just leave the one line here deprecating `refreshResources()` and point them to the new Forge methods.

The new methods can work like this:
```java
public void refreshResources(IResourceType... types)
{
    refreshResources(ReloadRequirements.include(types));
}

public void refreshResources(Predicate<IResourceType> resourcePredicate)
{
    SelectiveReloadStateHandler.INSTANCE.pushReloadRequirements(resourcePredicate);
    client.refreshResources();
    SelectiveReloadStateHandler.INSTANCE.popReloadRequirements();
}
```

That will reduce the patches in this file to just 1, and make the methods a little more readable.

Also, is there a situation where recursive reloads happen? I'm curious why there is a push/pop system.

Reply:
The push/pop system was mainly introduced on the off-chance that some mod may invoke a reload during a reload. Is this something that should be supported, or might it not be worth it?